File: java-sample/src/main/java/com/moengage/sample/java/FirebaseReceiver.java
package com.moengage.sample.java;

import androidx.annotation.NonNull;
import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;
import com.moengage.push.PushManager;

public class FirebaseReceiver extends FirebaseMessagingService {
  @Override public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
    super.onMessageReceived(remoteMessage);
    new SampleNotificationBuilder(getApplicationContext(),
        remoteMessage.getData()).buildAndShowNotification();
  }

  @Override public void onNewToken(@NonNull String token) {
    super.onNewToken(token);
    PushManager.getInstance().refreshToken(getApplicationContext(), token);
  }
}

File: kotlin-sample/src/main/java/com/moengage/sample/kotlin/FirebaseReceiver.kt
package com.moengage.sample.kotlin

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.moengage.push.PushManager

/**
 * @author Umang Chamaria
 * Date: 15/07/20
 */
class FirebaseReceiver : FirebaseMessagingService() {
    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)
        SampleNotificationBuilder(
            applicationContext,
            remoteMessage.data
        ).buildAndShowNotification();
    }

    override fun onNewToken(token: String) {
        super.onNewToken(token)
        PushManager.getInstance().refreshToken(applicationContext, token)
    }
}